crates/core/src/db/datastore/traits.rs

```diff
@@ -454,6 +97,16 @@ pub trait TxDatastore: DataRow + Tx {
         table_id: TableId,
         row_id: Self::RowId,
     ) -> Result<Option<Self::DataRef>>;
+
+    fn scan_st_tables<'a>(&'a self, tx: &'a Self::TxId) -> Result<Vec<StTableRow<String>>>;
+
+    fn scan_st_columns<'a>(&'a self, tx: &'a Self::TxId) -> Result<Vec<StColumnRow<String>>>;
+
+    fn scan_st_constraints<'a>(&'a self, tx: &'a Self::TxId) -> Result<Vec<StConstraintRow<String>>>;
+
+    fn scan_st_sequences<'a>(&'a self, tx: &'a Self::TxId) -> Result<Vec<StSequenceRow<String>>>;
+
+    fn scan_st_indexes<'a>(&'a self, tx: &'a Self::TxId) -> Result<Vec<StIndexRow<String>>>;
```

Comment:
I find those a little unfortunate:

- Do we really need this as public API?
- Why `scan_*` when `Vec` is returned?
- How will we decide which system tables can be collected, and ensure that new ones get a method here? Could we not pair st table ids with row types using associated types, and have a single method which works for all system tables?

Reply:
Done, I created a utility struct instead.